### PR TITLE
Ensure compatability with girder4 permission endpoints

### PR DESCRIPTION
**Pull request by naglepuff**
This pull request updates `multinetjs` to be compatible with the new Girder4 back end API. Specifically, interfaces defined in here now comply with the Django data model for users and permissions, and the `axios` calls to permission endpoints have been updated for compatibility.

Changes include:

1. Update to `UserSpec` interface
2. Update to `getWorkspacePermissions` and `setWorkspacePermissions` functions

### CODE REVIEW

**Comment by naglepuff** [comment]
`multinet-client` has a [PR](https://github.com/multinet-app/multinet-client/pull/183) that is dependant on this

File: src/index.ts [hunk 1]
```diff
@@ -68,6 +70,15 @@ export interface WorkspacePermissionsSpec {
   public: boolean;
 }
 
+export interface WorkspacePermissionRequestSpec {
+  /* In order to set permissions, the API only needs usernames */
+  owner: MinimumUserSpec;
+  maintainers: MinimumUserSpec[];
+  writers: MinimumUserSpec[];
+  readers: MinimumUserSpec[];
```
Comment (waxlamp) [inline src/index.ts]: If this is the only place `MinimumUserSpec` is needed, then it seems better to use `string` in place of it, and "promote" the strings into the proper structure needed by the endpoint after the data arrives inside the appropriate function. This is in the interest of keeping the calling sequences for the functions in question as human-centric and intuitive as possible.

If `MinimumUserSpec` is unavoidable, then I'd refactor its definition and that of `UserSpec` to use an `extends` relationship.

**Review by waxlamp — CHANGES_REQUESTED** [review]

**Review by AlmightyYakob — APPROVED** [review]
LGTM, tested locally

File: src/index.ts (hunk 1 above)
Comment (naglepuff) [reply src/index.ts]: Turns out that `MinimumUserSpec` is not needed. The most recent commit removes it.

**Pull request merged**